openstack/compute/v2/extensions/availabilityzones/results.go

```diff
@@ -1,8 +1,92 @@
 package availabilityzones
 
-// ServerAvailabilityZoneExt is an extension to the base Server result which
-// includes the Availability Zone information.
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/pagination"
+)
+
+// ServerExt is an extension to the base Server object
 type ServerAvailabilityZoneExt struct {
 	// AvailabilityZone is the availabilty zone the server is in.
 	AvailabilityZone string `json:"OS-EXT-AZ:availability_zone"`
 }
+
+type StateofService struct {
+	Active    bool      `json:"active"`
+	Available bool      `json:"available"`
+	UpdatedAt time.Time `json:"updated_at"`
+}
+
+// UnmarshalJSON to override default
+func (r *StateofService) UnmarshalJSON(b []byte) error {
+	type tmp StateofService
+	var s struct {
+		tmp
+		UpdatedAt gophercloud.JSONRFC3339MilliNoZ `json:"updated_at"`
+	}
+	err := json.Unmarshal(b, &s)
+	if err != nil {
+		return err
+	}
+	*r = StateofService(s.tmp)
+
+	r.UpdatedAt = time.Time(s.UpdatedAt)
+
+	return nil
+}
+
+type Services struct {
+	NovaConductor   StateofService `json:"nova-conductor"`
+	NovaConsoleauth StateofService `json:"nova-consoleauth"`
+	NovaNetwork     StateofService `json:"nova-network"`
+	NovaScheduler   StateofService `json:"nova-scheduler"`
+	NovaCompute     StateofService `json:"nova-compute"`
+	NovaCert        StateofService `json:"nova-cert"`
+}
```

Comment:
The above is hard-coding each of the nova services into Gophercloud. By doing this, each Host result will contain all of the above services even if they are not part of the availability zone. For example, this is the result of running this code in my test environment:

```go
                  "hosts": {
                    "openstack-acc-tests.novalocal": {
                      "nova-conductor": {
                        "active": false,
                        "available": false,
                        "updated_at": "0001-01-01T00:00:00Z"
                      },
                      "nova-consoleauth": {
                        "active": false,
                        "available": false,
                        "updated_at": "0001-01-01T00:00:00Z"
                      },
                      "nova-network": {
                        "active": false,
                        "available": false,
                        "updated_at": "0001-01-01T00:00:00Z"
                      },
                      "nova-scheduler": {
                        "active": false,
                        "available": false,
                        "updated_at": "0001-01-01T00:00:00Z"
                      },
                      "nova-compute": {
                        "active": true,
                        "available": true,
                        "updated_at": "2018-01-05T03:16:04Z"
                      },
                      "nova-cert": {
                        "active": false,
                        "available": false,
                        "updated_at": "0001-01-01T00:00:00Z"
                      }
                    }
                  },
                  "zoneName": "nova",
                  "zoneState": {
                    "available": true
                  }
                }
```

Instead of doing the above, I think we should do:

```go
type Service map[string]ServiceState
```

When I modify this PR to be the above, I now see this:

```go
                  "hosts": {
                    "openstack-acc-tests.novalocal": {
                      "nova-compute": {
                        "active": true,
                        "available": true,
                        "updated_at": "2018-01-05T03:26:34Z"
                      }
                    }
                  },
                  "zoneName": "nova",
                  "zoneState": {
                    "available": true
                  }
                }
```

Which I think is more accurate.

Please let me know if I am incorrect about this.